Angular/src/app/components/loader/loader.component.ts

```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-loader',
+  standalone: true,
+  imports: [],
+  templateUrl: './loader.component.html',
+  styleUrl: './loader.component.scss'
+})
+export class LoaderComponent {
+  @Input() isLoading: boolean = false;
```

Comment:
Can the LoaderComponent not inject the LoaderService and take care of this logic internally? That is usually the best practise.

Reply:
passing loader signal to component. otherwise we can do the Loader Interceptor as well but that will work on Api calls only, this is only to fire loader on custom triggers as well. can you please suggest the way?

Reply:
app component is not creating instance of service but you can see in above example every component need to create the instance. in my opinion current implementation is solid and good to go.

Reply:
both approaches are doing same way. signal seems more good in this usecase.

Reply:
in my opinion, let it be the current way we have, we will change if required explicitly. thanks

Reply:
or we can do is loader component can start listening to behaviourSubject() and creating instances of service to trigger the loader. are you fine with this approach? @sonukapoor

Reply:
You are not using `Signal`. You are using an `Input` only.

Reply:
![image](https://github.com/user-attachments/assets/6dda98db-162b-46b0-84a8-5282faf82ee4)

Reply:
Sure. Leave as is for now we can adjust later.

Reply:
we are passing `signal `value to `laoderComponent`